### PR TITLE
Define web-buffer class and move load-status, proxy, cookies and cert…

### PR DESCRIPTION
**Pull request by Ambrevar**
…ificates there.

### CODE REVIEW

File: source/buffer.lisp [hunk 1]
```diff
@@ -244,7 +255,15 @@ Must be one of `:always' (accept all cookies), `:never' (reject all cookies),
...
 
-(define-user-class buffer)
+(define-user-class web-buffer)
+
+(defmethod initialize-instance :after ((buffer web-buffer) &key)
+  (when (expand-path (cookies-path buffer))
+    (ensure-parent-exists (expand-path (cookies-path buffer)))))
+
+(export-always 'web-buffer-p)
+(defun web-buffer-p (buffer)
```
Comment (jmercouris) [inline source/buffer.lisp:265]: you could use a method that specializes on web-buffer and another one that just specializes on buffer as has been done for internal buffers

Comment (Ambrevar) [reply source/buffer.lisp:265]: I saw you did that, I just wanted to play with a different implementation style.  This one is shorter, I'm don't know if there is any drawback.

**Comment by Ambrevar** [comment]
@jmercouris Good to merge?

**Comment by jmercouris** [comment]
Indeed!

**Pull request merged**